jetbrains-core/src/software/aws/toolkits/jetbrains/services/ecr/actions/PushTagToRepositoryAction.kt

```diff
@@ -150,20 +209,155 @@ internal class PushToEcrDialog(
+            }
+        }
+
+        val browseExtension = ExtendableTextComponent.Extension.create(
+            AllIcons.General.OpenDisk, AllIcons.General.OpenDiskHover,
+            message("ecr.dockerfile.configuration.add")
+        ) {
+            val listener = object : TextBrowseFolderListener(
+                FileChooserDescriptorFactory.createSingleFileDescriptor(DockerFileType.DOCKER_FILE_TYPE),
+                project
+            ) {
+                init {
+                    myTextComponent = textComponent
+                }
+
+                override fun getInitialFile() = this@PushToEcrDialog.project.guessProjectDir()
+
+                override fun onFileChosen(chosenFile: VirtualFile) {
+                    val settings = dockerRunConfigurationFromPath(this@PushToEcrDialog.project, chosenFile.presentableName, chosenFile.path)
+                    // open dialog for user
+                    RunDialog.editConfiguration(
+                        project, settings,
+                        ExecutionBundle.message("run.dashboard.edit.configuration.dialog.title")
+                    )
+                    rebuildRunConfigurationComboBoxModel(runConfigModel)
+                }
+            }
+
+            runInEdt {
+                listener.run()
+            }
+        }
+
+        // extensions from right to left
+        textComponent.setExtensions(browseExtension, editExtension)
+    }
+
+    private fun rebuildRunConfigurationComboBoxModel(model: CollectionComboBoxModel<DockerRunConfiguration>) {
+        val configs = RunManager.getInstance(project).getConfigurationsList(DockerCloudType.getRunConfigurationType())
+            .filterIsInstance<DockerRunConfiguration>()
+            .filter {
+                // there are multiple types of Docker run configurations. only accept Dockerfile for now
+                // "image" and "compose" both seem like they only make sense as run-only configurations
+                it.deploymentSource.type == DockerFileDeploymentSourceType.getInstance()
+            }
+
+        model.replaceAll(configs)
+        model.selectedItem = configs.firstOrNull()
+    }
+
+    private fun selectedRepo() = remoteRepos.selected() ?: throw IllegalStateException("repository uri was null")
+
+    suspend fun getPushRequest() = when (type.ordinal) {
+        BuildType.LocalImage.ordinal -> ImageEcrPushRequest(
+            dockerServerRuntime.await(),
+            localImage?.imageId ?: throw IllegalStateException("image id was null"),
+            selectedRepo(),
+            remoteTag
+        )
+
+        BuildType.Dockerfile.ordinal -> DockerfileEcrPushRequest(
+            runConfiguration ?: throw IllegalStateException("run configuration was null"),
+            selectedRepo(),
+            remoteTag
+        )
+
+        else -> throw IllegalStateException()
+    }
+
+    private enum class BuildType {
+        LocalImage, Dockerfile
+    }
+}
+
+// TODO: unify utils with other PR
```

Comment:
to fulfill this TODO, just dump it in the same package and I'll fix it on the other side when this merges